src/libs/actions/PersonalDetails.ts

```diff
@@ -276,9 +281,172 @@ function openPublicProfilePage(accountID: number) {
+        lastName: 'Smith',
+        guidedSetupData: JSON.stringify([{
+            reportID: parentReportID,
+            reportActionID: reportCommentAction.reportActionID,
+            reportComment: reportCommentText,
+            url: `${CONST.CLOUDFRONT_URL}/videos/intro-1280.mp4`,
+            duration: 55,
+            width: 1280,
+            height: 960,
+            thumbnailUrl: `${CONST.CLOUDFRONT_URL}/images/expensify__favicon.png`,
+            type: 'video',
+        }, {
+            reportID: parentReportID,
+            reportActionID: reportCommentAction2.reportActionID,
+            reportComment: reportCommentText2,
+            type: 'message',
+        }, {
+            parentReportActionID: optimisticAddCommentReport.reportAction.reportActionID,
+            parentReportID,
+            taskReportID: optimisticTaskReport.reportID,
+            createdTaskReportActionID: optimisticTaskCreatedAction.reportActionID,
+            title: optimisticTaskReport.reportName,
+            description: optimisticTaskReport.description,
+            assigneeChatReportID,
+            type: 'task',
+            task: 'trackExpense',
+        }, {
+            reportID: optimisticTaskReport.reportID,
+            reportActionID: reportCommentAction3.reportActionID,
+            reportComment: reportCommentText3,
+            type: 'message',
+        }, {
+            parentReportActionID: optimisticAddCommentReport2.reportAction.reportActionID,
+            parentReportID,
+            taskReportID: optimisticTaskReport2.reportID,
+            createdTaskReportActionID: optimisticTaskCreatedAction2.reportActionID,
+            title: optimisticTaskReport2.reportName,
+            description: optimisticTaskReport2.description,
+            assigneeChatReportID,
+            // The `createWorkspace` tasks needs an action ID because we'll immediately complete it in the backend.
+            completedTaskReportActionID: NumberUtils.rand64(), // TODO: Use the ID of a proper optimistic action ID instead, see `Task.completeTask`
+            type: 'task',
+            task: 'createWorkspace',
+        }, {
+            reportID: optimisticTaskReport2.reportID,
+            reportActionID: reportCommentAction4.reportActionID,
+            reportComment: reportCommentText4,
+            type: 'message',
+        }]),
+        engagementChoice: 'newDotValue',
+    };
+
+    const optimisticData2: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`,
+            value: {
+                // Message 1 (video)
+                [reportCommentAction.reportActionID ?? '']: reportCommentAction as ReportAction,
+                // Message 2 (plain text)
+                [reportCommentAction2.reportActionID ?? '']: reportCommentAction2 as ReportAction,
+            },
+        }, {
+            // This doesn't seem to work, not sure why
```

Comment:
could you just use merge? sometime using mix of set and merge behaves werid

Reply:
Vit! 😂 
![image](https://github.com/Expensify/App/assets/16747822/b2fef0ec-e3ee-4763-b7db-f0a8afc86e1d)

Reply:
I have just added a single comment, not a review 😂 idk why Github is this way